### ISSUE TITLE
Verify that the debugSetup () and debugDraw () functions are correct.

### ISSUE THREAD

**Issue by kalwalt** [augmentmy-world/artoolkitX.js#4]
In this code: 

 ```javascript
 try {
        //arController.setLogLevel(1);
        arController.start().then( () => {
            console.log("start done");
            var trackableId = arController.addTrackable(trackable);
            interval = setInterval(function() {
                arController.process(video);
            }, 13);
            ar1 = arController;
        });
        arController.debugSetup();
    }
    catch (e) {
        console.log(e);
    }
```
the code produce two canvas the left one with the videoframe and the other should be the debugSetup(),
What happens if you add a debugDraw() after debugSetup()? You will get an error:

```shell
TypeError: Cannot convert undefined or null to object
    at Uint8ClampedArray.set (<anonymous>)
    at ARController.debugDraw (artoolkitX.api.js:760)
    at simple_video.html:48
    at Object.Module.onRuntimeInitialized (artoolkitx.js:6)
    at doRun (artoolkitx.js:6)
    at run (artoolkitx.js:6)
    at runCaller (artoolkitx.js:6)
    at removeRunDependency (artoolkitx.js:6)
    at receiveInstance (artoolkitx.js:6)
    at artoolkitx.js:6

```
see the https://github.com/artoolkit/jsartoolkit5/blob/master/examples/simple_image.html example and  https://github.com/artoolkit/jsartoolkit5/blob/master/examples/threejs_from_scratch.html  example for a comparison.

**Comment by kalwalt** [comment]
With the debugSetup() function i will aspect the same result as in the jsartoolkit5 example https://github.com/artoolkitx/jsartoolkit5/blob/master/examples/simple_rtc.html
see the screenshot:
![Schermata del 2019-08-21 14-28-05](https://user-images.githubusercontent.com/1275858/63432139-d6028e00-c420-11e9-924a-450134d90172.png)
not exactly the same with artoolkitx.js
![Schermata del 2019-08-21 14-55-00](https://user-images.githubusercontent.com/1275858/63433614-d3556800-c423-11e9-9ffe-0c915e0fb250.png)
with arController.setLogLevel(1);

**Comment by kalwalt** [comment]
i partiallly solved it was caused because we need to fill the data in the canvas (this.ctx) with the same size, we need to refill it see the code:

```js
ARController.prototype.debugDraw = function () {
        var videoMalloc = artoolkitXjs.videoMalloc;
        var debugBuffer = new Uint8ClampedArray(artoolkitXjs.HEAPU8.buffer, videoMalloc.lumaFramePointer, videoMalloc.framesize);
        var id = new ImageData(new Uint8ClampedArray(this.canvas.width*this.canvas.height*4), this.canvas.width, this.canvas.height);
    		for (var i=0, j=0; i<debugBuffer.length; i++, j+=4) {
    			var v = debugBuffer[i];
    			id.data[j+0] = v;
    			id.data[j+1] = v;
    			id.data[j+2] = v;
    			id.data[j+3] = 255;
    		}
        this.ctx.putImageData(id, 0, 0);
```

**Comment by kalwalt** [comment]
solved with PR #15
